### PR TITLE
Feature/40 login with google

### PR DESCRIPTION
**Pull request by jimibilibob**

### CODE REVIEW

File: RoutesApp_iosTests/AuthTests.swift [hunk 1]
```diff
@@ -62,4 +82,56 @@ class AuthTests: XCTestCase {
...
+            case .failure:
+                XCTAssertFalse(authManager.isValidCredential)
+            }
+        }
+    }
+
+    func testGoogleSignInValidToken() {
+        let authManager = signupViewmodel.authManager as? MockAuthManager ?? MockAuthManager()
+        authManager.isValidToken = false
+        authManager.signInWithGoogle(target: UIViewController()) { result in
```
[the remark this answers is not in the document]
Comment (jimibilibob) [reply RoutesApp_iosTests/AuthTests.swift:119]: ok, I will check it

File: RoutesApp_ios/modules/authentication/signup/viewmodels/SignupViewModel.swift [hunk 2]
```diff
@@ -67,4 +68,68 @@ class SignupViewModel {
...
+                self.onError?(error.localizedDescription)
+            }
+        }
+    }
+
+    private func getUser(_ typeLogin: UserTypeLogin, email: String, completion: @escaping ((_ user: User?) -> Void)) {
+        userManager.getUsers { result in
+            switch result {
+            case .success(let users):
+                let user = users.filter {$0.email == email && $0.typeLogin == typeLogin.rawValue} .first
```
[the remark this answers is not in the document]
Comment (jimibilibob) [reply RoutesApp_ios/modules/authentication/signup/viewmodels/SignupViewModel.swift]: Yes, it's important in order to allow the authentication flow with firebase.

File: RoutesApp_ios/repository/network/managers/FirebaseAuthManager.swift [hunk 3]
```diff
@@ -29,4 +34,34 @@ class FirebaseAuthManager: AuthProtocol {
             }
         }
     }
+
+    func signInWithGoogle(target: UIViewController, completion: @escaping (Result<(credential: NSObject, email: String), Error>) -> Void) {
+        GIDSignIn.sharedInstance.signOut()
+        guard let clientID = FirebaseApp.app()?.options.clientID else { return }
+        let config = GIDConfiguration(clientID: clientID)
+        GIDSignIn.sharedInstance.signIn(with: config, presenting: target) { (user, error) in
+            if error != nil { return completion(.failure(error!)) }
```
[the remark this answers is not in the document]
Comment (jimibilibob) [reply RoutesApp_ios/repository/network/managers/FirebaseAuthManager.swift]: Dewraping is been used now

File: Podfile [hunk 4]
```diff
@@ -9,6 +9,7 @@ target 'RoutesApp_ios' do
   pod 'SwiftLint', '0.47.1'
   pod 'SVProgressHUD', '2.2.5'
   pod 'IQKeyboardManagerSwift', '6.2.1'
+  pod 'GoogleSignIn'
```
Comment (jmacboy) [inline Podfile]: Missing pod version

**Review by jmacboy — CHANGES_REQUESTED** [review]
just that and you´re good to go

File: Podfile (hunk 4 above)
Comment (jimibilibob) [reply Podfile]: ok, I will add it

Comment (jimibilibob) [reply Podfile]: Added GoogleSignIn version

**Review by jmacboy — APPROVED** [review]
Approved but check bitrise

**Pull request merged**